core/dbt/task/run.py

```diff
@@ -430,7 +431,10 @@ def before_run(self, adapter, selected_uids: AbstractSet[str]):
         with adapter.connection_named("master"):
             required_schemas = self.get_model_schemas(adapter, selected_uids)
             self.create_schemas(adapter, required_schemas)
-            self.populate_adapter_cache(adapter, required_schemas)
+            if get_flags().CACHE_SELECTED_ONLY is True:
+                self.populate_adapter_cache(adapter, required_schemas)
+            else:
+                self.populate_adapter_cache(adapter)
```

Comment:
IMO it makes more sense to have the `CACHE_SELECTED_ONLY` conditional check here, rather than in `populate_adapter_cache`. The behavior of `populate_adapter_cache` should just be: If I'm handed a specific set of schemas, cache those schemas; otherwise, cache all schemas defined in the manifest.

(Relevant to the current PR because the `clone` task wants to cache _more_ schemas than just the ones defined in _this_ manifest — we also want to cache the _other_ manifest's schemas)